cpp/dolfinx/fem/Expression.h

```diff
@@ -86,23 +92,18 @@ class Expression
   /// Evaluate the expression on cells
   /// @param[in] active_cells Cells on which to evaluate the Expression
   /// @param[out] values A 2D array to store the result. Caller
-  /// responsible for correct sizing which should be (num_cells,
-  /// num_points * value_size columns).
+  /// is responsible for correct sizing which should be (num_cells,
+  /// num_points * value_size * num_all_argument_dofs columns).
```

Comment:
I don't understand what `num_all_argument_dofs` is.

Wondering if this is the best layout now that it's a 4D tensor. Would it be better to pass e.g. a span of the correct size and document the layout when reinterpreted into a 4D tensor?